adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -478,10 +481,13 @@ public final void uploadEventMessage(final RoutingContext ctx, final String tena
     private void doUploadMessage(final RoutingContext ctx, final String tenant, final String deviceId,
             final Buffer payload, final String contentType, final Future<MessageSender> senderTracker, final String endpointName) {
 
+        final Integer qosHeader = ctx.request().getHeader(HEADER_QOS_LEVEL) == null ? null : Integer.parseInt(ctx.request().getHeader(HEADER_QOS_LEVEL));
```

Comment:
please use our code formatter rules to be found under `/eclipse` ...
this line seems to be too long

Reply:
thanks

Reply:
`Integer.parseInt(...)`
And what happens if the header is not containing an integer value that can be parsed?

Can you please add some handling for this? Otherwise this would generate lots of ugly exceptions inside the adapter when a client misbehaves...

Reply:
sure, thanks